### ISSUE TITLE
[dcl.fct] packs in *parameter-type-list*

### ISSUE THREAD

**Issue by cor3ntin** [cplusplus/CWG#629]
in decl.fct p4, we say

After producing the list of parameter types, any top-level [cv-qualifier](https://eel.is/c++draft/dcl.decl.general#nt:cv-qualifier)s modifying a parameter type are deleted when forming the function type[.](https://eel.is/c++draft/dcl.fct#4.sentence-4) The resulting list of transformed parameter types and the presence or absence of the ellipsis **or a function parameter pack** is the function's [parameter-type-list](https://eel.is/c++draft/dcl.fct#def:parameter-type-list).

Emphasis mine.

It's not clear what we mean by "or a function parameter pack".
A function can have a number of parameter packs in arbitrary position, and the fact that only the trailing pack is a deduced context is irrelevant when we are not deducing anything.

Are these functions equivalent?
```cpp
void f(auto..., auto) {}
void f(auto..., auto) {}
```

Proposed resolution (which needs further massaging)

After producing the list of parameter types, any top-level [cv-qualifier](https://eel.is/c++draft/dcl.decl.general#nt:cv-qualifier)s modifying a parameter type are deleted when forming the function type[.](https://eel.is/c++draft/dcl.fct#4.sentence-4) The resulting list of transformed parameter types <ins>and whether each corresponding parameter-declaration is a function parameter pack</ins> and the presence or absence of the ellipsis <del>or a function parameter pack</del> is the function's [parameter-type-list](https://eel.is/c++draft/dcl.fct#def:parameter-type-list).

**Comment by jensmaurer** [comment]
CWG2942

**Comment by cor3ntin** [comment]
> I think whether a given parameter declaration is a pack or not is covered by the preceding wording. We can just strike the part about parameter packs where we talk about ellipsis.

I am not sure this is correct. the "packness" is part of the declaration, not the type. So presumably, there needs to be some wording to propagate that.

**Comment by jensmaurer** [comment]
I disagree.  (Note that we're at the template level here, so "type" might be a bit more than what the rest of the type system, after instantiation, knows about.)

"The type of each parameter (including function parameter packs) is determined from its own parameter-declaration (9.3)'"

The packness is part of the parameter-declaration; if we feel something is amiss, [dcl.meaning.general] would need amendment somewhere. Suggestions welcome.
